src/actions/courseActions.js

```diff
@@ -0,0 +1,53 @@
+ import * as types from './actionTypes';
+ import CourseApi from '../api/mockCourseApi';
+ import {beginAjaxCall, ajaxCallError} from './ajaxStatusActions';
+
+export const createCourseSuccess=(course)=>{
+    return{type:types.CREATE_COURSE_SUCCESS, course};
+};
+
+export const loadCoursesSuccess = (courses)=> {
+  return {type: types.LOAD_COURSES_SUCCESS, courses};
+};
+
+export const loadCourses = () => {
+    return function (dispatch ) {
+      dispatch(beginAjaxCall());
+      return CourseApi.getAllCourses().then(courses => {
+        dispatch(loadCoursesSuccess(courses));
+      }).catch(error => {
+        throw(error);
+      });
+    };
+  };
+  export const updateCourseSuccess = (course)=> {
+    return {type: types.UPDATE_COURSE_SUCCESS, course};
+  };
+
+    export const saveCourse = (course)=> {
```

Comment:
Nice trick using the course id as the logic for updating or saving.
But this function name does not properly define the function the code is executing. something like
```saveOrUpdateCourse``` might make more sense